src/test/java/jssc/junit/rules/VirtualPortRule.java

```diff
@@ -99,7 +99,7 @@ public void evaluate() throws Throwable {
         } finally {
           // stop socat
           for (final Future<?> process : VirtualPortRule.this.processes) {
-            process.cancel(false);
+            process.cancel(true);
```

Comment:
Looks like the only code that tries to kill the background process is a `catch` block for `InterruptedException`. I am not sure why we would want to set `mayInterruptIfRunning` to `false` like this - am I missing something?

Reply:
This was copied verbatim from a PR that was written by a developer that no longer contributes to this project, so I'm not not sure what the historical decision was based off of.

Reply:
> I am not sure why we would want to set mayInterruptIfRunning to false like this - am I missing something?

I honestly didn't understand the code, just tested it.  The only thing I can think of was that the author's original intention was to use a one-to-many initially.  As it stands, I **think** interrupting it is the only way it will cancel the parent process.